### PR TITLE
Web Editor

### PR DESCRIPTION
**Pull request by Lorenzo0111**
This pull request adds support for a web editor using the Gson serialization. It already works but it can't be merged because we have to create the web part.

### CODE REVIEW

**Comment by JustOneMoreBlock** [comment]
I was thinking of deploying an update for 1.17.1 and then releasing it. This doesn't need this at the moment. Does it?

**Comment by Lorenzo0111** [comment]
> I was thinking of deploying an update for 1.17.1 and then releasing it. This doesn't need this at the moment. Does it?

No, this will require some other days. You can start releasing the 1.17 update

**Comment by Lorenzo0111** [comment]
**Status Update:** The plugin backend is ready. The NametagEditor ( web ) is not ready. You can view a todo list in Lorenzo0111/NametagEditor#1

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # NametagEdit
 
 [![Dev Builds](https://img.shields.io/badge/Jenkins-Development%20Builds-lightgrey.svg)](http://ci.nametagedit.com/job/NametagEdit/)
-[![Support](https://img.shields.io/badge/Minecraft-1.7--1.16-green.svg)](documentation/Support)
```
Comment (Vinetos) [inline README.md:4]: It shouldn't be in this MR as this is notpart of the feature :}

File: src/main/java/com/nametagedit/plugin/NametagCommand.java [hunk 2]
```diff
@@ -182,6 +188,33 @@ private void sendUsage(CommandSender sender) {
...
+                }
+
+                String code = args[1];
+                EditorSession session = EditorSession.from(handler.getPlugin(), sender, code);
+                session.retrieveData();
+                NametagMessages.EDITOR_IMPORT.send(sender);
+
+            } catch (IOException e) {
+                NametagMessages.EDITOR_ERROR.send(sender);
+                e.printStackTrace();
```
Comment (Vinetos) [inline src/main/java/com/nametagedit/plugin/NametagCommand.java:213]: Maybe use the plugin logger ?

File: src/main/java/com/nametagedit/plugin/web/EditorSession.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    private static final URL PASTE;
+    private final NametagEdit plugin;
+    private final CommandSender player;
+    @Setter
+    @Nullable private String code;
+
+    static {
+        URL paste = null;
+
+        // todo edit
```
Comment (Vinetos) [inline src/main/java/com/nametagedit/plugin/web/EditorSession.java:45]: load from config file ?

**Review by Vinetos — COMMENTED** [review]
As it is in dev, I've added some commentaries to the code :D

File: src/main/java/com/nametagedit/plugin/web/EditorSession.java (hunk 3 above)
Comment (Lorenzo0111) [reply src/main/java/com/nametagedit/plugin/web/EditorSession.java:45]: This is a work in progress, when the web editor will be ready I’ll change it

**Pull request merged**